### ISSUE TITLE
Inference GeoLayoutLM

### ISSUE THREAD

**Issue by GendalfSeriyy** [AlibabaResearch/AdvancedLiterateMachinery#36]
Hello! Thanks a lot for the great work!
I have a question regarding model inference. As far as I can see, the model forward requires block boxes to be fed. But in order to get the block boxes, you must first get them, for example, using LayoutLM.
Is it possible to run an inference of model with only an image, word boxes, and their corresponding text as input? Could you provide the code for the inference?

**Comment by ccx1997** [comment]
Hello! In fact, text bounding boxes may have several levels, as in [1][2]. You can get block boxes by a hierarchical text detection model or layout analysis (not neccessary to use LayoutLM, since just the block box is okay).

In the academic benchmarks, the block boxes are often given. SOTA methods (e.g. LayoutLMv3) tend to exploit the block information as well to improve the performance furthermore. We believe it is an important visual clue for the layout representation.

Of course, it is also possible to try GeoLayoutLM with only the word boxes if the block boxes are hard to obtain. You may try to treat each word box as also a block box that only has a single word within it, so the input arguments can be made in the same way. But I cannot promise the result in this case, especially when the finetuning samples are not enough.

Overall, you can either manage to have the block boxes, or try to treat word boxes as block boxes. The preparation for the model input arguments is also available in the modelscope (multi-lingual version) in a more elegant way.

[1] https://rrc.cvc.uab.es/?ch=18&com=introduction
[2] Long et.al., Towards End-to-End Unified Scene Text Detection and Layout Analysis. In CVPR 2022.

**Comment by mike-grayhat** [comment]
The NN looks very promising, but I'm facing the same issue - the bboxes I'm getting from an ocr engine doesn't correspond to entity units, so there's no way to make proper linking in some cases. I would even say that block aggregation is better be incapsulated into the language model itself, since it's often relies on block semantic value, so an ocr engine based on visual cues will sometimes get it wrong. As a solution I'm planning to add another heads for bio entity linking within a block to predict containing words and their relative order.

**Comment by HoomanKhosravi** [comment]
@mike-grayhat Hi Mike, any success with this issue?

**Comment by mike-grayhat** [comment]
@HoomanKhosravi I'm at the evaluation phase of GeoLayoutLM for my dataset, and I'd decided to start with a simple solution first - just use one word blocks, do bio tagging and extend block links accordingly. It works, but performance is almost identical to my bros-based solution, which might be due to many factors.

**Comment by mike-grayhat** [comment]
I also have an idea that since GeoLayoutLM has builtin bio tagging for inblock words ([here](https://github.com/AlibabaResearch/AdvancedLiterateMachinery/blob/c33064be5574d39579df60c907a1f1c0d652b0a3/DocumentUnderstanding/GeoLayoutLM/lightning_modules/data_modules/vie_dataset.py#L273) and [here](https://github.com/AlibabaResearch/AdvancedLiterateMachinery/blob/c33064be5574d39579df60c907a1f1c0d652b0a3/DocumentUnderstanding/GeoLayoutLM/lightning_modules/geolayoutlm_vie_module.py#L25)), which are basically labels, maybe it's actually possible with some hacking to make GeoLayoutLM work without relying on block information, especially for zero/few shot inference.

**Comment by ccx1997** [comment]
> Of course, it is also possible to try GeoLayoutLM with only the word boxes if the block boxes are hard to obtain. You may try to treat each word box as also a block box that only has a single word within it, so the input arguments can be made in the same way. But I cannot promise the result in this case, especially when the finetuning samples are not enough.

With insufficient training data, the segment/block box should be very useful to the final performance because there exists more prior infomation.

> performance is almost identical to my bros-based solution

In the case, I guess the SER may be almost identical to your bros-based solution, but the RE should still outperform much? If not, maybe the input block box is also important to modeling of the geometric relations of entities.

BTW, it is efficient by making use of block information when dealing with long documents (num_tokens >> 512).

**Comment by dyh** [comment]
咋样了，有推理代码了吗？
